sqlglot/dialects/snowflake.py

```diff
@@ -438,17 +434,17 @@ def _parse_alter_table_swap(self) -> exp.SwapTable:
 
         def _parse_location(self) -> exp.LocationProperty:
             self._match(TokenType.EQ)
+            return self.expression(exp.LocationProperty, this=self._parse_location_path())
 
-            parts = [self._parse_var(any_token=True)]
+        def _parse_location_path(self) -> exp.Var:
+            parts = [self._advance_any(ignore_reserved=True)]
```

Comment:
why do we need this first one?

Reply:
because the first token in the path token sequence can be "disconnected" from its previous tokens and so we won't enter the loop and crash in `exp.var("")`.